test/boost/sstable_3_x_test.cc

```diff
@@ -5107,10 +5107,10 @@ SEASTAR_TEST_CASE(test_sstable_reader_on_unknown_column) {
     auto mt = make_lw_shared<replica::memtable>(write_schema);
     mt->apply(partition);
     for (auto index_block_size : {1, 128, 64*1024}) {
-        tmpdir dir;
+        auto _ = env.tempdir().make_sweeper();
```

Comment:
> ... so the first iteration might see garbage that other tests left. Can't this cause problems?

In fact first iteration starts over fresh new tempdir, so it's naturally empty

Reply:
> A different thing you could have done is to replace make_sweeker() by a method temp_subdir() which creates an (empty) subdirectory in the temp directory and returns it - and when the object is destroyed the subdirectory is deleted. This not only solves the problem I asked about (the new subdirectory starts out empty) it also allows running multiple tests in parallel, in case that's something you'll ever want to do (though I doubt it).

Good idea, indeed. btw. There are several cases that create a set/vector of tempdirs to do something. Currently there's no way to switch them onto sstables::test_env's tempdir, but with temp_subdir() it will likely work. :)

Reply:
> > ... so the first iteration might see garbage that other tests left. Can't this cause problems?
> 
> In fact first iteration starts over fresh new tempdir, so it's naturally empty

Unless I'm misunderstanding something, the cql_env doesn't clear the temporary directory between tests, so if one test leaves stuff in the directory by mistake, it will be there also for the next test. This is not terrible if tests are written to assume there may be garbage in this directory (and use unique names for files - like we do for sstables), but if some test assumes that the directory starts out empty, it might be disappointed.